cdap-common/src/main/java/co/cask/cdap/common/io/Locations.java

```diff
@@ -344,15 +344,19 @@ public static Location getParent(Location location) {
   /**
    * Given a locationPath which is derived from URI.getPath() and a locationFactory
    * this method generates a Location with URI generated from the given Absolute path
+   * The given absolutePath is expected to contain basePath if any.
    * @param locationFactory locationFactory to create Location from given Path
    * @param absolutePath Path to be used for Location
    * @return Location resulting from absolute locationPath
    */
   public static Location getLocationFromAbsolutePath(LocationFactory locationFactory, String absolutePath) {
-    URI basePathURI = URI.create(locationFactory.create("").toURI().getPath());
-    URI locationURI = URI.create(absolutePath);
-    URI relativePathURI = basePathURI.relativize(locationURI);
-    return locationFactory.create(relativePathURI);
+    URI homeURI = locationFactory.getHomeLocation().toURI();
+    try {
+      return locationFactory.create(new URI(homeURI.getScheme(), homeURI.getAuthority(), absolutePath, null));
```

Comment:
See a Twill PR here: https://github.com/apache/twill/pull/28
Would a similar problem be here?